app/src/test/java/ch/hikemate/app/navigation/NavigationActionsTest.kt

```diff
@@ -0,0 +1,114 @@
+import org.mockito.Mockito.`when`
+import org.mockito.kotlin.any
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.eq
+
+class NavigationActionsTest {
+
+  private lateinit var navigationDestination: NavDestination
+  private lateinit var navHostController: NavHostController
+  private lateinit var navigationActions: NavigationActions
+
+  private lateinit var navGraph: NavGraph
+  private lateinit var navController: TestNavHostController
+  private lateinit var context: Context
+  private var navOptionsCaptor = argumentCaptor<NavOptionsBuilder.() -> Unit>()
+
+  @Before
+  fun setUp() {
+    context = mock(Context::class.java)
+    navigationDestination = mock(NavDestination::class.java)
+    navHostController = mock(NavHostController::class.java)
+    navigationActions = NavigationActions(navHostController)
+    navGraph = mock(NavGraph::class.java)
+    `when`(navHostController.graph).thenReturn(navGraph)
+    navController = spy(TestNavHostController(context))
+  }
+
+  @Test
+  fun navigateToCallsController() {
+    navigationActions.navigateTo(TopLevelDestinations.OVERVIEW)
+    verify(navHostController)
+        .navigate(eq(Route.OVERVIEW), org.mockito.kotlin.any<NavOptionsBuilder.() -> Unit>())
+    navigationActions.navigateTo(Screen.MAP)
+    verify(navHostController).navigate(Screen.MAP)
+  }
+
+  @Test
+  fun goBackCallsController() {
+    navigationActions.goBack()
+    verify(navHostController).popBackStack()
+  }
+
+  @Test
+  fun currentRouteWorksWithDestination() {
+    `when`(navHostController.currentDestination).thenReturn(navigationDestination)
+    `when`(navigationDestination.route).thenReturn(Route.OVERVIEW)
+
+    assertThat(navigationActions.currentRoute(), `is`(Route.OVERVIEW))
+  }
+
+  @Test
+  fun currentRoute_isEmptyWhenNoDestination() {
+    `when`(navHostController.currentDestination).thenReturn(null)
+    assertThat(navigationActions.currentRoute(), `is`(""))
+  }
+
+  @Test
+  fun currentRoute_isEmptyWhenNoRoute() {
+    `when`(navHostController.currentDestination).thenReturn(navigationDestination)
+    `when`(navigationDestination.route).thenReturn(null)
+    assertThat(navigationActions.currentRoute(), `is`(""))
+  }
+
+  @Test
+  fun navigateToNotAuth_respectsProperties() {
+    navigationActions.navigateTo(TopLevelDestinations.OVERVIEW)
+
+    navHostController.navigate(currentRoute_isEmptyWhenNoRoute())
+    verify(navHostController).navigate(eq(Route.OVERVIEW), navOptionsCaptor.capture())
+    val navOptions = navOptions(navOptionsCaptor.firstValue)
+
+    assertThat(navOptions.shouldRestoreState(), `is`(true))
+    assertThat(navOptions.isPopUpToInclusive(), `is`(true))
+    assertThat(navOptions.shouldPopUpToSaveState(), `is`(true))
+    assertThat(navOptions.shouldLaunchSingleTop(), `is`(true))
+  }
+
+  @Test
+  fun navigateToAuth_respectsProperties() {
+    navigationActions.navigateTo(TopLevelDestination(Route.AUTH, Icons.Default.AccountBox, "Auth"))
```

Comment:
I suggested a change in `NavigationActions.kt` that would change this to:
```suggestion
    navigationActions.navigateTo(TopLevelDestinations.AUTH)
```

Reply:
Auth is not a TopLevelDestination(as suggested by @davidholzwarth in another comment) since it won't appear in the SideBarButton.kt, so this change is not  possible.

Reply:
As of my understanding, `TopLevelDestination` should be all the start of any route, not tabs to appear in `SideBarButton.kt`. We could create a `SidebarTabs = [TopLevelDestination.Overview, ...)` to then choose which to show in the sidebar.

I still argue we should change this. Maybe we can discuss with the team if you still don't agree

Reply:
We agreed on my side of view